asserts/database.go

```diff
@@ -303,6 +336,13 @@ func (db *Database) Add(assert Assertion) error {
 
 	err := db.Check(assert)
 	if err != nil {
+		if ufe, ok := err.(*UnsupportedFormatError); ok {
+			_, err := ref.Resolve(db.Find)
+			if err != nil && err != ErrNotFound {
+				return err
+			}
+			return &UnsupportedFormatError{Ref: ufe.Ref, Format: ufe.Format, Update: err == nil}
```

Comment:
Maybe copy the existing error and explicitly set Update on the copy to ensure anything added in the future is carried across.